nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -562,11 +562,18 @@ sub addProfile {
             next;
         }
         my $date = strftime("%F", localtime(lstat($link)->mtime));
-        my $version =
-            -e "$link/nixos-version"
-            ? readFile("$link/nixos-version")
-            : basename((glob(dirname(Cwd::abs_path("$link/kernel")) . "/lib/modules/*"))[0]);
-        addEntry("NixOS - Configuration " . nrFromGen($link) . " ($date - $version)", $link);
+        my $nixos_version = readFile("$link/nixos-version");
```

Comment:
Hrm, maybe it would be nice to add a way to replace NixOS with a vanity name on top of the nixos-version symlink? 
Possibly out of scope since it requires adding an option to the module and behavior to match in other generators though, but if you want to do it it might be a good idea.
 Could also allow for an additional custom string for additional infos